File: Java/Civitas/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.Random;

public class Dado {
    private Random random=new Random();
    private int ultimoResultado;
    private Boolean debug;
    private static final Dado instance=new Dado();
    private static final int SalidaCarcel=5;
    
    private Dado(){
        ultimoResultado=0;
        debug=false;
    }
    
    static public Dado getInstance(){
        return instance;
    }
    
    int tirar(){
        if(debug){
            ultimoResultado=1;
        }
        else{
            ultimoResultado=random.nextInt(6)+1;
        }
        return ultimoResultado;
    }
    
    Boolean salgoDeLaCarcel(){
        return tirar()>=SalidaCarcel;
    }
    
    int quienEmpieza(int n){
        Random casual=new Random();
        return casual.nextInt(n);
    }
    
    void setDebug(Boolean d){
        debug=d;
		Diario.getInstance().ocurreEvento("Se ha " + (d?"activado":"desactivado")+" el modo debug del dado.");
    }
    
    public int getUltimoResultado(){
        return ultimoResultado;
    }
}

File: Ruby/CivitasR/lib/Dado.rb
#encoding:utf-8

require 'singleton'
module Civitas

  class Dado
    include Singleton
    attr_reader :ultimoResultado
    
    @@SalidaCarcel = 5

    def initialize
      @ultimoResultado = -1
      @debug = false
    end

    def tirar
      @ultimoResultado = @debug? 1:rand(1..6)
      return @ultimoResultado
    end

    def salgoDeLaCarcel
      return @@SalidaCarcel >= rand(1..6)
    end

    def quienEmpieza(n)
      return rand(n)
    end

    def setDebug(d)
      @debug = d
      Diario.instance.ocurre_evento("Dado -- modo debug: " + d.to_s)
    end


  end
end
